### ISSUE TITLE
assertEmailCount stopped working after upgrading to 6.0.5

### ISSUE THREAD

**Issue by garak** [symfony/symfony#45580]
### Symfony version(s) affected

6.0.5

### Description

In a project with Symfony 6.0.3, after upgrading to 6.0.5 I got this error while executing tests (I found that, in general, the problem arises as soon as the container is retrieved):

```
LogicException: Could not find service "test.service_container". Try updating the "framework.test" config to "true".

/myproj/vendor/symfony/framework-bundle/Test/KernelTestCase.php:98
/myproj/vendor/symfony/framework-bundle/Test/MailerAssertionsTrait.php:121
/myproj/vendor/symfony/framework-bundle/Test/MailerAssertionsTrait.php:25
/myproj/tests/Controller/DefaultControllerTest.php:136

Caused by
Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: The "kernel" service is synthetic, it needs to be set at boot time before it can be used.
```
If I execute `bin/console deb:conf framework` I can see that the `test` configuration is actually set to `true`
If I execute `bin/console de:cont --parameter=kernel.environment` I get (as expected) "test".

The only way I found so far to get it working again is adding this to my composer.json: 

```json
"conflict": {
        "symfony/framework-bundle": "6.0.5"
    },
```

### How to reproduce

I cannot reproduce the problem on a fresh installation. 

### Possible Solution

Commenting out `$container->reset()` in https://github.com/symfony/framework-bundle/blob/6.0/KernelBrowser.php#L155 solves the problem.

**Comment by VincentLanglet** [comment]
I have the same issue on https://github.com/sonata-project/SonataUserBundle/pull/1513

**Comment by IonBazan** [comment]
It also occurs to me during test execution. Probably introduced in #45479

**Comment by xabbuh** [comment]
Can you create a small example application that allows to reproduce your issue?

**Comment by garak** [comment]
> Can you create a small example application that allows to reproduce your issue?

Well, no.
I already tried with a fresh project and the problem is not reproduced.
Anyway, I can say that I have it on two different projects

**Comment by IonBazan** [comment]
@xabbuh @garak You can use current version of https://github.com/symfony/demo and run following commands:
```shell
composer install #installs v6.0.1
vendor/bin/simple-phpunit # Passes
composer update symfony/framework-bundle #installs v6.0.5
vendor/bin/simple-phpunit # Fails
```

You may see that only the tests that require to make several requests are failing. For example, `BlogControllerTest ::testAdminNewPost` submits a form.

**Comment by nicolas-grekas** [comment]
Thanks for the reproducer.
Does calling disableReboot() work around the issue for my info?

**Comment by dsech** [comment]
This also happens on one of my projects after upgrading to 5.4.5. Calling `disableReboot()` solves the issue in my case.

**Comment by PhilETaylor** [comment]
Just another vote to say that Calling `disableReboot()` not he `$client` in 5.4.5 fixes my unit tests.

**Comment by kbond** [comment]
~Related: https://github.com/symfony/symfony/pull/45581~ Separate problem.

**Comment by jbator** [comment]
Confirm that this also happens in symfony 5 after upgrading to 5.4.5

**Comment by nicolas-grekas** [comment]
Will be fixed by #45595

You can call `disableReboot()` as a workaround.

**Comment by COil** [comment]
If you still want to upgrade, fix de framework-bundle version:

    "symfony/framework-bundle": "v6.0.4"
